Scripts/TagGenerator.py

```diff
@@ -0,0 +1,211 @@
+#
+# Module for automatically tagging a commit with a release version.
+#
+# Copyright (c) Microsoft Corporation
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+#
+
+import argparse
+import re
+from git import Repo
+from datetime import datetime
+import time
+
+parser = argparse.ArgumentParser()
+
+parser.add_argument("-r", "--repo", default=".",
+                    help="Path to the repo directory.")
+parser.add_argument("-b", "--base", type=str, required=True,
+                    help="The base release version. This must be provided")

```

Comment:
The original intention was to keep major for the "breaking", but without the need to concatinate the EKII release tag version with the breaking changes version. It isn't really as clear as I'd hoped though so I think you are right. I will change to major.minor.patch, but semantically the minor will be breaking changes. 

Alternatively we could conform to the versioning standard we use on the git repros, but I fear this might add some confusion on the consumers here since the distinction between features and bug fixes may not be useful.